packages/graphiql/src/utility/mergeAst.ts

```diff
@@ -6,39 +6,25 @@
  */
 import { Kind } from 'graphql/language/kinds';
 import {
-  FragmentDefinitionNode,
-  Location,
   DocumentNode,
-  DefinitionNode,
-  NameNode,
-  VariableDefinitionNode,
-  NamedTypeNode,
-  DirectiveNode,
-  SelectionSetNode,
+  FragmentDefinitionNode,
+  OperationDefinitionNode,
 } from 'graphql';
 
-interface MutableDocumentNode extends DocumentNode {
-  definitions: ReadonlyArray<DefinitionNode>;
-}
+type FragmentMap = { [key: string]: OperationDefinitionNode };
 
-interface MutableFragmentDefinitionNode {
-  kind: 'FragmentDefinition' | 'InlineFragment';
-  readonly loc?: Location;
-  readonly name: NameNode;
-  // Note: fragment variable definitions are experimental and may be changed
-  // or removed in the future.
-  readonly variableDefinitions?: ReadonlyArray<VariableDefinitionNode>;
-  readonly typeCondition: NamedTypeNode;
-  readonly directives?: ReadonlyArray<DirectiveNode>;
-  readonly selectionSet: SelectionSetNode;
-}
-
-function resolveDefinition(fragments, obj) {
-  let definition = obj;
-  if (definition.kind === Kind.FRAGMENT_SPREAD) {
-    definition = fragments[definition.name.value];
-  }
+function resolveDefinition(
+  fragments: FragmentMap,
+  obj: OperationDefinitionNode,
+): OperationDefinitionNode {
+  // if we are dealing with a fragment spread then we have to look up that definition in the
+  // fragment definition map
+  const definition =
+    (obj as any).kind === Kind.FRAGMENT_SPREAD && obj.name
```

Comment:
I'm not a huge fan of this `any` but the current implementation relies on `kind` which makes the type checking logic a little tricky. Since we wanted to avoid major refactors, I opted to just keep it as is to merge the typescript stuff asap.